manifests/plugin/threshold.pp

```diff
@@ -2,12 +2,16 @@
 class collectd::plugin::threshold (
   $ensure   = 'present',
   $interval = undef,
+  Array[Collectd::Threshold::Type]   $types   = [],
+  Array[Collectd::Threshold::Plugin] $plugins = [],
```

Comment:
This is a breaking change, right? Because of the custom datatype, an empty array isn't allowed anymore and users now have to pass data to the new parameters, correct? Is that desired? If so, I need to mark it as backwards-incompatible.

Reply:
I don't think so… I am quite new to the threshold plugin of collectd, but as far as I experienced, the puppet-collectd module allowed to *enable* the `threshold` module but *not to* set a configuration, making it somewhat useless (the reason why I worked on this).

The default values of `[]` for `$types`, `$plugins` and `$hosts` imply that if a user previously had enabled the `threshold` module, if he does not change his configuration, the module is just loaded as it was before, without setting any thresholds (if the user used something external to the puppet-collectd module to setup a configuration, I guess this change should have no impact).

As a matter of facts, the first unit tests do not set `$types`, `$plugins` and `$hosts`:

https://github.com/smortex/puppet-collectd/blob/29a87d1a44acb80b238e22ca2168032025cb5c9f/spec/classes/collectd_plugin_threshold_spec.rb#L71-L99